GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1223,25 +1223,19 @@ bool ConfirmAndStageAllUnstaged()
 
                     // Option 1: there are no staged files, but there are unstaged files. Most probably user forgot to stage them.
                     string text = Unstaged.IsFilterActive ? _noFilesStagedCommitAllFilteredUnstagedOption.Text : _noFilesStagedCommitAllUnstagedOption.Text;
-                    TaskDialogCommandLinkButton lnkStageAndCommit = new(text, enabled: !Unstaged.IsEmpty)
-                    {
-                        Tag = 1
-                    };
+                    TaskDialogCommandLinkButton lnkStageAndCommit = new(text, enabled: !Unstaged.IsEmpty);
                     lnkStageAndCommit.Click += (s, e) =>
                     {
                         mustStageAll = true;
                     };
                     page.Buttons.Add(lnkStageAndCommit);
 
                     // Option 2: the user just wants to make an empty commmit
-                    TaskDialogCommandLinkButton lnkEmptyCommit = new(_noFilesStagedMakeEmptyCommitOption.Text)
-                    {
-                        Tag = 2
-                    };
+                    TaskDialogCommandLinkButton lnkEmptyCommit = new(_noFilesStagedMakeEmptyCommitOption.Text);
                     page.Buttons.Add(lnkEmptyCommit);
 
                     TaskDialogButton result = TaskDialog.ShowDialog(Handle, page);
-                    if (result.Tag is not int)
+                    if (result == TaskDialogButton.Cancel)
```

Comment:
`if (result == TaskDialogButton.Cancel)`
to
`if (result is TaskDialogButton.Cancel)`

Reply:
It will not compile. `TaskDialogButton.Cancel` is not a constant value.

Reply:
IMO `is null` is ok, for everything else it looks awkward.